web_client/actions.jsx

```diff
@@ -529,14 +535,16 @@ export const submitLoginForm = (form) => promiseAction(
     let result;
     return (
       rest
-        .login(login, password)
+        .logout()
+        .then(() => rest.login(login, password))
         .then((user) => dispatch(setCurrentUser(user, user.token.token)))
         .then((user) => (result = user))
         .then(() => dispatch(closeDialog()))
         .then(() => result)
         .catch(
           ({ responseJSON: { message } }) => (
             dispatch(setDialogError('login', message))
+              .then(() => dispatch(loginAnonymousUser()))
           )
```

Comment:
I also just realized that the catch handler needs to return `Promise.reject(arg)` where arg is the argument passed to it.  This also needs to be done to propagate the error to the parent promise chain.